tls/s2n_server_extensions.c

```diff
@@ -202,21 +202,26 @@ int s2n_server_extensions_recv(struct s2n_connection *conn, struct s2n_blob *ext
 
         switch (extension_type) {
         case TLS_EXTENSION_SERVER_NAME:
+            S2N_ERROR_IF(s2n_is_tls13_enabled(), S2N_ERR_BAD_MESSAGE);
             GUARD(s2n_recv_server_server_name(conn, &extension));
             break;
         case TLS_EXTENSION_RENEGOTIATION_INFO:
             GUARD(s2n_recv_server_renegotiation_info_ext(conn, &extension));
             break;
         case TLS_EXTENSION_ALPN:
+            S2N_ERROR_IF(s2n_is_tls13_enabled(), S2N_ERR_BAD_MESSAGE);
```

Comment:
I believe this needs to be `conn->actual_protocol_version == S2N_TLS13` as tls 1.2 connections may be served after turning tls13 mode

Reply:
ack, my tired eyes missed that.